src/main/java/Main.java

```diff
@@ -414,6 +419,32 @@ private static File extractFromJar(String resource, String fileName, String suff
         return tmp;
     }
 
+    /**
+     * Search contents to delete in a folder that match with some patterns.
+     * @param folder folder where the contents are.
+     * @param patterns patterns that identifies the contents to search.
+     * @throws IOException in case of error deleting contents.
+     */
+    private static void deleteContentsFromFolder(File folder, final String...patterns) throws IOException {
+        final File[] files = folder.listFiles(new FilenameFilter() {
+            @Override
+            public boolean accept(final File dir, final String name) {
+                for (String pattern : patterns) {
+                    if(name.matches(pattern)){
```

Comment:
it would make sense to precompile regular expressions to boost the performance

Reply:
Better to just inline this method, so it is only called once and with a single regexp. The code is more complicated than is actually needed to fix the issue.